package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -175,18 +175,22 @@ end
 
 -- Generates a (hopefully) unique MAC address
 -- The parameter defines the ID to add to the MAC address
---
--- IDs defined so far:
--- 0: client0; WAN
--- 1: mesh0
--- 2: owe0
--- 3: wan_radio0 (private WLAN); batman-adv primary address
--- 4: client1; LAN
--- 5: mesh1
--- 6: owe1
--- 7: wan_radio1 (private WLAN); mesh VPN
-function M.generate_mac(i)
-	if i > 7 or i < 0 then return nil end -- max allowed id (0b111)
+
+M.radio_mac_offsets = {
```

Comment:
Hmm, maybe move `radio_mac_offsets` into `wireless.lua`, make both offset tables `local`. `wireless.lua` would be the only place that deals with radio numbers then.

`M.generate_mac` would just have a `func` argument then, `generate_mac_by_idx` is made public again and called both by `generate_mac` and `get_wlan_mac`.

Reply:
Then I would import wireless in gluon_bat0.sh, 330-gluon-mesh-batman-adv-mac-addresses, 500-mesh-vpn and gluon_wired.sh

which do not have anything to do with wireless.. Otherwise this suggestion would be fine to me

Reply:
Why would you import that? Only `radio_mac_offsets` would be moved, `if_mac_offsets` should remain in `util.lua`.

Reply:
Thanks, totally agree. That's a good refactoring step.